### DISCUSSION TITLE
buefy-next on nuxt3

### DISCUSSION THREAD

**Discussion by ardeay** [buefy/buefy#4086] [Q&A] [answered]
Is there a nuxt3 plugin or any documentation to best implement it?

Thank you

**Comment by wesdevpro** [comment, marked as answer]
Hey @ardeay that is such a great question! Because we are still tied up with migrating Buefy to Vue3, for the time being, we won't be moving the next-buefy plugin. However, for the time being, you can use a plugin like this in your Nuxt plugin folder 
// plugins/buefy.ts (or.js)
```javascript
import Buefy from 'buefy'
import 'buefy/dist/buefy.css'

export default defineNuxtPlugin((nuxtApp) => {
    nuxtApp.vueApp.use(Buefy, {})
})
```

Please let me know if I can help with anything else. You can contact me more directly on our discord server(https://discord.buefy.org).
Thanks

**Reply by dagnelies** [reply]
Really great. Thanks for that tip, it should land in the docs!

**Reply by dagnelies** [reply]
For the sake of completeness, you can also declare global dialogs/toasts that way:

```ts
export default defineNuxtPlugin((nuxtApp) => {
    const appWithBuefy = nuxtApp.vueApp.use(Buefy, {})

    // @ts-ignore
    const dialog = new DialogProgrammatic(appWithBuefy);
    nuxtApp.provide('dialog', dialog)
    
    // @ts-ignore
    const toast = new ToastProgrammatic(appWithBuefy);
    nuxtApp.provide('toast', toast)
})
```

And import them where you need it as follows:

```ts
const { $dialog, $toast } = useNuxtApp();
//...
$dialog.alert(...)
$toast.open(...)
```
